### PR TITLE
[#163889865] Checkout Manager, Build Metadata, and Slack Notifications are shared concerns

### PR DESCRIPTION
**Pull request by austinworks**

### CODE REVIEW

**Comment by austinworks** [comment]
[#163889865]

**Comment by austinworks** [comment]
please squash this if you approve/merge

File: src/com/omadahealth/CheckoutManager.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        println err.toString()
+        SCMFailureException typedError = new SCMFailureException(workerName: 'problem node')
+        throw typedError
+      }
+
+      actions()
+    }
+  }
+
+  def make_workspace(Closure actions) {
```
Comment (jimmyatwork) [inline src/com/omadahealth/CheckoutManager.groovy]: Did you mean to use snakecase for this method and `make_workspace_with_checkout`?

File: src/com/omadahealth/RuntimeLogManager.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+  def collateLogs(logNames, collationTarget) {
+    try {
+      steps.timeout(time: 10, unit: 'SECONDS') {
+        for(j=0; j < logNames.length; j++) {
+          steps.unstash logNames[j]
+        }
+      }
+      def collation = steps.libraryResource 'com/omadahealth/scripts/collation.sh'
```
Comment (jimmyatwork) [inline src/com/omadahealth/RuntimeLogManager.groovy]: Pretty nice that you can use embedded scripts!

File: src/com/omadahealth/CheckoutManager.groovy [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  def steps
+  CheckoutManager(steps, nodeType='ubuntu-18-04') {
+    this.steps = steps
+    this.nodeType = nodeType
+  }
+
+  def reconnectNode(workerName) {
+    for (aWorker in hudson.model.Hudson.instance.slaves) {
+      if ( aWorker.name == workerName) {
+        if ( aWorker.getComputer().isOnline()) {
```
Comment (jimmyatwork) [inline src/com/omadahealth/CheckoutManager.groovy]: Above two `if` statements have extra spaces after the parentheses.

File: src/com/omadahealth/CheckoutManager.groovy (hunk 1 above)
Comment (austinworks) [reply src/com/omadahealth/CheckoutManager.groovy]: hmm that is a good thought

File: src/com/omadahealth/RuntimeLogManager.groovy (hunk 2 above)
Comment (austinworks) [reply src/com/omadahealth/RuntimeLogManager.groovy]: yeah it's a cool idea, gonna see if that's actually working...

**Pull request merged**